### PR TITLE
Handle property groups round trip vCard <-> jCard according to spec

### LINKED ISSUE #411: Better handling of vCard property groups
Property names can have a prefix indicating the group they are associated with. For example:
```
BEGIN:VCARD
VERSION:3.0
UID:4445-07DE0202-0206-0B34-FF16-00A2C
PRODID:-//Apple Inc.//iOS 11.2.6//EN
FN:Mr. Smith
item1.EMAIL;type=INTERNET,pref:main@email.com
EMAIL;type=INTERNET,WORK:work@email.com
END:VCARD
```

Here we have both `item1.EMAIL` and `EMAIL`. There are two problems with how ical.js handles these:
1. `comp.getFirstProperty('email')` (and other such functions) will only return the second email which is confusing and makes handling these much harder.
2. The field name is not correctly processed so a few things like field type will be incorrect.

### PR DESCRIPTION
**Pull request by castaway**
Based on: https://github.com/mozilla-comm/ical.js/pull/440

Updated to use group conversion rules: https://tools.ietf.org/html/rfc7095#section-3.3.1.2.1

Fix #411

### CODE REVIEW

File: lib/ical/stringify.js [hunk 1]
```diff
@@ -108,7 +108,15 @@ ICAL.stringify = (function() {
...
-    var line = name;
+    var groupName = params['group'];
+    var line;
+    if (groupName) {
+      line = groupName + "." + name;
+    } else {
+      line = name;
+    }
+
+    delete params.group;
```
Comment (darktrojan) [inline lib/ical/stringify.js:119]: This shouldn't be deleted as the jCal is then changed for future operations. Instead the loop below should do nothing if `paramName` is `group`.

File: lib/ical/stringify.js [hunk 2]
```diff
@@ -108,7 +108,15 @@ ICAL.stringify = (function() {
     var jsName = property[0];
     var params = property[1];
 
-    var line = name;
+    var groupName = params['group'];
+    var line;
+    if (groupName) {
+      line = groupName + "." + name;
```
Comment (darktrojan) [inline lib/ical/stringify.js:114]: `groupName` should be made upper-case, since `name` is upper-case.

File: lib/ical/stringify.js [hunk 3]
```diff
@@ -108,7 +108,15 @@ ICAL.stringify = (function() {
     var jsName = property[0];
     var params = property[1];
 
-    var line = name;
+    var groupName = params['group'];
```
Comment (kewisch) [inline lib/ical/stringify.js:111]: This stringifies both jCard and jCal. The group param does not work that way in jCal though. We may need to have a hook from the designSet that allows augmenting the property name.

File: lib/ical/parse.js [hunk 4]
```diff
@@ -232,9 +232,20 @@ ICAL.parse = (function() {
     var multiValue = false;
     var structuredValue = false;
     var propertyDetails;
+    var splitName;
+    var ungroupedName;
+
+    // fetch the ungrouped part of the name
+    if (name.indexOf('.') !== -1) {
```
Comment (kewisch) [inline lib/ical/parse.js:239]: Likewise to comment below, we need to differentiate iCalendar vs vCard here by using the designSet.

**Comment by kewisch** [comment]
@darktrojan Is this fixed by #512 ?

**Comment by darktrojan** [comment]
Yes.

**Pull request closed without merging**